src/components/LawProposals.vue

```diff
@@ -12,34 +15,68 @@ export default defineComponent({
 
   data() {
     return {
+      laws: [] as Law[],
       selectedLaw: undefined as LawId | undefined,
+      fadeingLaw: undefined as LawId | undefined,
     }
   },
 
-  props: {
-    proposedLaws: Array as PropType<Law[]>,
+  mounted() {
+    const e = this.$el as HTMLElement
+    e.style.setProperty("--transitiontime", TRANSITION_TIME + "s")
+  },
+
+  computed: {
+    ...mapGetters(["proposedLaws"]),
   },
 
   methods: {
-    select(lawId: LawId) {
+    select(lawId: LawId | undefined) {
       this.selectedLaw = lawId
     },
 
     accept(lawId: LawId) {
-      this.store.dispatch("acceptLaw", { lawId })
+      this.fadeingLaw = lawId
+      setTimeout(() => this.store.dispatch("acceptLaw", { lawId }), TRANSITION_TIME * 1000)
     },
 
     decline(lawId: LawId) {
-      this.store.dispatch("rejectLaw", { lawId })
+      this.fadeingLaw = lawId
+      setTimeout(() => this.store.dispatch("rejectLaw", { lawId }), TRANSITION_TIME * 1000)
     },
   },
+
+  watch: {
+    proposedLaws(newLaws: Law[]) {
+      const laws: Law[] = [...this.laws]
+      const addedLaws: Law[] = newLaws.filter(nl => !laws.some(ol => ol.id === nl.id))
+
+      for (var i = 0; i < laws.length; i++) {
+        if (newLaws.some(nl => nl.id === laws[i].id)) {
+          continue
+        }
+        if (addedLaws.length === 0) {
+          laws.splice(i)
+          continue
+        }
+        laws[i] = addedLaws.shift() as Law
+      }
+      laws.push(...addedLaws)
+      this.laws = laws
+    }
+  }
 })
 </script>
 
 <template>
   <div class="ProposedLaws">
-    <div v-for="(law, index) in proposedLaws" :key="index" class="Law" :class="{ selected: law.id === selectedLaw }">
-      <div @click="select(law.id)">
+    <div
+      v-for="(law, index) in laws"
+      :key="index"
+      class="Law"
+      :class="{ selected: law.id === selectedLaw, fading: law.id === fadeingLaw }"
```

Comment:
Wenn du die Vergleiche als `computed` fields anlegst unter dem Namen der CSS-Klasse, die sie bewirken sollen, kannst du das Ganze sehr kompakt schreiben:

```suggestion
      :class="{ selected, fading }"
```

Reply:
So direkt geht das aber noch nicht, oder? (Ich probier mal lokal)

Reply:
In `computed` komme ich ja nicht an `law.id` ran...

Reply:
Guter Punkt. Dann bringt's natürlich nichts.
Die "saubere" Lösung wäre, für jedes Listenelement eine eigene Vue-Komponente anzulegen. Ist aber mE etwas oversized, insofern lassen wir's doch einfach so.

Reply:
Ist jetzt noch wieder anders geworden. Ich habs tatsächlich in `computed` bekommen - allerdings gibt das eine funktion zurück... Das hat vermutlich keine Vorteile gegenüber `methods`.

Das mit der eigenen Komponente hatte ich auch schon überlegt. Daran, dass alle Methods eine `lawId`als Parameter bekommen sieht man ja, dass wir da nicht so richtig objekt-orientiert sind. Ich denke, dass wir in naher Zukunft sowieso noch dran müssen. Dann sollten wir das wohl gleich als Erstes machen.